src/pybind/mgr/dashboard/frontend/src/app/shared/api/host.service.spec.ts

```diff
@@ -0,0 +1,32 @@
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import { TestBed } from '@angular/core/testing';
+
+import { configureTestBed } from '../unit-test-helper';
+import { HostService } from './host.service';
+
+describe('HostService', () => {
+  let service: HostService;
+  let httpTesting: HttpTestingController;
+
+  configureTestBed({
+    providers: [HostService],
+    imports: [HttpClientTestingModule]
+  });
+
+  beforeEach(() => {
+    service = TestBed.get(HostService);
+    httpTesting = TestBed.get(HttpTestingController);
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  it('should call list', () => {
+    service.list().then();
+    const req = httpTesting.expectOne('api/host');
+    expect(req.request.method).toBe('GET');
+    req.flush(['foo', 'bar']);
```

Comment:
This line can be removed as the outcome is not used or it has to be transformed to something like this to use it.

```
  it(
    'should call list',
    fakeAsync(() => {
      let x = undefined;
      service.list().then((y) => (x = y));
      const req = httpTesting.expectOne('api/host');
      expect(req.request.method).toBe('GET');
      req.flush(['foo', 'bar']);
      tick();
      expect(x).toEqual(['foo', 'bar']);
      httpTesting.verify();
    })
  );
```

Reply:
Test should be also used, besides testing production code, for developers to understand the code.

The promise is only resolved if you use fakeAsync or mock it out.
If you want the promise to be resolved - which currently isn't the case, you have to do it this way. 

Currently you can remove `.then()` in line 26 as it is never executed, because of this you can remove the flush too.

You should not rely on the coverage report that everything is tested, as it isn't, the return statement is never resolved inside the promise.